strawberryfields/configuration.py

```diff
@@ -14,139 +14,214 @@
 r"""
 This module contains the :class:`Configuration` class, which is used to
 load, store, save, and modify configuration options for Strawberry Fields.
+
+.. warning::
+
+    The following configuration options are available:
+    :doc:`/introduction/configuration`.
+
 """
-import os
 import logging as log
+import os
 
 import toml
 from appdirs import user_config_dir
 
 log.getLogger()
 
-
-DEFAULT_CONFIG = {
-    "api": {
-        "authentication_token": "",
-        "hostname": "localhost",
-        "use_ssl": True,
-        "port": 443,
-        "debug": False}
+DEFAULT_CONFIG_SPEC = {
+   "api": {
+        "authentication_token": (str, ""),
+        "hostname": (str, "localhost"),
+        "use_ssl": (bool, True),
+        "port": (int, 443),
+    }
 }
 
-BOOLEAN_KEYS = ("debug", "use_ssl")
+BOOLEAN_KEYS = {"use_ssl"}
+INTEGER_KEYS = {"port"}
```

Comment:
Now that the above `DEFAULT_CONFIG_SPEC` specifies the types, are these still needed?

Reply:
Oh not at all, thanks for that!